stdlib/LinearAlgebra/src/symmetric.jl

```diff
@@ -902,26 +902,28 @@ symmetricpart!(A::AbstractMatrix, uplo::Symbol=:U) =
     Symmetric(_hermorsympart!(symmetricpart, transpose, A, char_uplo(uplo)), uplo)
 
 """
-    hermitianpart(A, uplo=:U) -> Hermitian
+    hermitianpart(A, uplo=:U, full=false) -> Hermitian
 
 Return the Hermitian part of the square matrix `A`, defined as `(A + A') / 2`,
 as a [`Hermitian`](@ref) matrix. For real matrices `A`, this is also known as the symmetric
-part of `A`. The optional argument `uplo` controls whether the upper
-(`uplo = :U`) or lower (`uplo = :L`) triangle of the underlying matrix is explicitly
-filled. The opposite triangle is then only implicitly filled via the [`Hermitian`](@ref)
-view, which, for real matrices, is equivalent to a [`Symmetric`](@ref) view.
+part of `A`. The first optional argument `uplo` controls whether the upper
+(`uplo = :U`) or lower (`uplo = :L`) triangle of the matrix underlying the [`Hermitian`](@ref)
+view is explicitly filled. Unless the second optional argument `full=true` (in which case the
+full underlying matrix is filled with the Hermitian part), the opposite triangle is then
+only implicitly filled via the [`Hermitian`](@ref) view. For real matrices, the latter is
+equivalent to a [`Symmetric`](@ref) view.
 
 See also [`hermitianpart!`](@ref) for the corresponding in-place operation.
 
 !!! compat "Julia 1.10"
     This function requires Julia 1.10 or later.
 """
-hermitianpart(A::AbstractMatrix{T}, uplo=:U) where {T} =
-    hermitianpart!(copy_similar(A, typeof(one(T) / 2)), uplo)
-hermitianpart(x::Number) = real(x)
+hermitianpart(A::AbstractMatrix{T}, uplo=:U, full::Bool = false) where {T} =
+    hermitianpart!(copy_similar(A, typeof(one(T) / 2)), uplo, full)
+hermitianpart(x::Number, uplo=:U, full::Bool=false) = real(x)
```

Comment:
This seems like a strange API — why should you need to pass `uplo` if you are using `full`?

Could it be `uplo=:full`?  Either that or `full` should be a keyword argument.